src/main/scala/no/gdl/bookapi/repository/TranslationRepository.scala

```diff
@@ -51,40 +53,33 @@ trait TranslationRepository {
       val limit = pageSize.max(1)
       val offset = (page.max(1) - 1) * pageSize
 
-      val results = select(t.result.bookId)
+      val result =
+        select(countAllBeforeLimiting, t.result.bookId)
         .from(Translation as t)
         .where.eq(t.language, language.toString)
         .append(getSorting(sortDef))
         .limit(limit).offset(offset)
         .toSQL
-        .map(_.long(1)).list().apply()
+        .map(row => (row.long(1), row.long(2))).list().apply()
 
-      SearchResult[Long](results.length, page, pageSize, language, results)
+      SearchResult[Long](result.map(_._1).headOption.getOrElse(0), page, pageSize, language, result.map(_._2))
     }
 
     def bookIdsWithLanguageAndLevel(language: LanguageTag, readingLevel: Option[String], pageSize: Int, page: Int, sortDef: Sort.Value)(implicit session: DBSession = ReadOnlyAutoSession): SearchResult[Long] = {
       val limit = pageSize.max(1)
       val offset = (page.max(1) - 1) * pageSize
 
-      val num_matching = select(sqls.count)
-        .from(Translation as t)
-        .where
-        .eq(t.language, language.toString)
-        .and(
-          sqls.toAndConditionOpt(readingLevel.map(l => sqls.eq(t.readingLevel, l))))
-        .toSQL.map(_.long(1)).single().apply()
-
-      val results = select(t.result.bookId)
+      val result = select(countAllBeforeLimiting, t.result.bookId)
```

Comment:
It's supposed to be a bit cheaper to do this in one query instead of two, but my main motivation for doing this is to count the actual things that we ask for, and not having to keep the two queries in sync if either either of them changes.